### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by anntzer**
Three independent cleanup commits:

- in axisartist: Mostly just unwrapping long lists of arguments and other expressions into a single line where possible.
- in contour: Shorten contour._make_paths_from_contour_generator.
- in dviread: Move most one-liner opcode parsers into lambdas with the behavior specified by comments, which seems clearer than spreading them out into proper functions with plenty of intervening blank space.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/mpl_toolkits/axisartist/axislines.py [hunk 1]
```diff
@@ -326,27 +313,18 @@ def __init__(self, axes):
...
             _api.warn_external(
                 "'new_fixed_axis' explicitly requires the axes keyword.")
             axes = self.axes
         if axis_direction is None:
             axis_direction = loc
-        helper = FixedAxisArtistHelperRectilinear(axes, loc)
-        axisline = AxisArtist(axes, helper, offset=offset,
-                              axis_direction=axis_direction)
-        return axisline
+        return AxisArtist(axes, FixedAxisArtistHelperRectilinear(axes, loc, nth_coord),
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/mpl_toolkits/axisartist/axislines.py:324]: oops, crossed PRs...

File: lib/matplotlib/dviread.py [hunk 2]
```diff
@@ -129,60 +129,25 @@ def glyph_name_or_index(self):
...
+    s4=lambda dvi, delta: dvi._arg(4, signed=True),
+    # slen: Read (opcode - minimum) bytes as a signed number, or None if delta is None.
+    slen=lambda dvi, delta: dvi._arg(delta, signed=True) if delta else None,
+    # slen1: Read (opcode - minimum + 1) bytes as a signed number.
+    slen1=lambda dvi, delta: dvi._arg(delta + 1, signed=True),
+    # slen1: Read (opcode - minimum + 1) bytes as an unsigned number.
+    ulen1=lambda dvi, delta: dvi._arg(delta + 1, signed=False),
+    # slen1: Read (opcode - minimum + 1) bytes as an unsigned number if less
+    # than 4 bytes, as a signed number if 4 bytes.
+    olen1=lambda dvi, delta: dvi._arg(delta + 1, signed=(delta == 3)),
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/dviread.py:149]: Not really, switched it.

**Review by QuLogic — APPROVED** [review]

**Review by ksunden — COMMENTED** [review]
Just some comment clean ups

File: lib/matplotlib/dviread.py [hunk 3]
```diff
@@ -129,60 +129,25 @@ def glyph_name_or_index(self):
...
+    u4=lambda dvi, delta: dvi._arg(4, signed=False),
+    # s4: Read 4 bytes as a signed number.
+    s4=lambda dvi, delta: dvi._arg(4, signed=True),
+    # slen: Read delta bytes as a signed number, or None if delta is None.
+    slen=lambda dvi, delta: dvi._arg(delta, signed=True) if delta else None,
+    # slen1: Read (delta + 1) bytes as a signed number.
+    slen1=lambda dvi, delta: dvi._arg(delta + 1, signed=True),
+    # slen1: Read (delta + 1) bytes as an unsigned number.
+    ulen1=lambda dvi, delta: dvi._arg(delta + 1, signed=False),
+    # slen1: Read (delta + 1) bytes as an unsigned number if less than 4 bytes,
```
Comment (ksunden) [inline lib/matplotlib/dviread.py:147]:
```suggestion
    # ulen1: Read (delta + 1) bytes as an unsigned number.
    ulen1=lambda dvi, delta: dvi._arg(delta + 1, signed=False),
    # olen1: Read (delta + 1) bytes as an unsigned number if less than 4 bytes,
```

Comment (anntzer) [reply lib/matplotlib/dviread.py:147]: oops, fixed.

**Review by ksunden — APPROVED** [review]

**Pull request merged**